app/src/sharedTest/java/org/oppia/android/app/topic/TopicFragmentTest.kt

```diff
@@ -358,19 +359,79 @@ class TopicFragmentTest {
     }
   }
 
+  @Test
+  fun testTopicFragment_withStoryId_clickOnPracticeTab_configChange_showsSameTabAndItsContent() {
+    launchTopicPlayStoryActivityIntent(
+      internalProfileId,
+      FRACTIONS_TOPIC_ID,
+      FRACTIONS_STORY_ID_0
+    ).use {
+      testCoroutineDispatchers.runCurrent()
+      clickTabAtPosition(position = PRACTICE_TAB_POSITION)
+      testCoroutineDispatchers.runCurrent()
+      matchStringOnListItem(
+        recyclerView = R.id.topic_practice_skill_list,
+        itemPosition = 0,
+        targetViewId = R.id.master_skills_text_view,
+        stringToMatch = "Master These Skills"
+      )
+      onView(isRoot()).perform(orientationLandscape())
+      testCoroutineDispatchers.runCurrent()
+      verifyTabTitleAtPosition(position = PRACTICE_TAB_POSITION)
+      matchStringOnListItem(
+        recyclerView = R.id.topic_practice_skill_list,
+        itemPosition = 0,
+        targetViewId = R.id.master_skills_text_view,
+        stringToMatch = "Master These Skills"
+      )
+    }
+  }
+
+  /**
+   * Creates TopicActivity Intent without a storyId
+   */
   private fun createTopicActivityIntent(internalProfileId: Int, topicId: String): Intent {
     return TopicActivity.createTopicActivityIntent(
       ApplicationProvider.getApplicationContext(), internalProfileId, topicId
     )
   }
 
+  /**
+   * Creates TopicActivity Intent with a storyId
+   */
+  private fun createTopicPlayStoryActivityIntent(
+    internalProfileId: Int,
+    topicId: String,
+    storyId: String
+  ): Intent {
+    return TopicActivity.createTopicPlayStoryActivityIntent(
+      ApplicationProvider.getApplicationContext(), internalProfileId, topicId, storyId
+    )
+  }
+
+  /**
+   * Launches TopicActivity without a storyId.
+   * This simulates opening a topic from All topics list.
```

Comment:
is it correct this simulates All Topics, as I think we have multiple things like Promoted stories, Recently Played, Coming Soon, View All.

@rt4914 please correct me if I am missing something here.

Reply:
This is correct.

For explanation:
1. Click on any item present in `Home Carousel` -> Opens `Topic-Lessons tab`
2. Click on any item present inside `RecentlyPlayedActivity` -> Opens exploration directly
3. Click on any item present on home screen in `All topics` section -> Opens `Topic Info Tab`.